BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1461,7 +1461,7 @@ public void BeehiveDetonation()
                 pBullet.timeElapsedSinceCurrentSpeedWasAdjusted = 0;
                 pBullet.timeToLiveUntil = 2000 / bulletVelocity * 1.1f + Time.time;
                 Vector3 firedVelocity = VectorUtils.GaussianDirectionDeviation(currentVelocity.normalized, subMunitionType.subProjectileDispersion > 0 ? subMunitionType.subProjectileDispersion : (subMunitionType.subProjectileCount / BDAMath.Sqrt(GetDragAdjustedVelocity().magnitude / 100))) * (GetDragAdjustedVelocity().magnitude + subMunitionType.bulletVelocity); //more subprojectiles = wider spread, higher base velocity = tighter spread
-                pBullet.currentVelocity = firedVelocity; //if submunitions have additional vel, would need modifications to ModuleWeapon's CPA calcs to offset targetPos by -targetVel * (submunitionVelocity / proximityDetonationdist)
+                pBullet.currentVelocity = BDKrakensbane.FrameVelocityV3f + firedVelocity; //if submunitions have additional vel, would need modifications to ModuleWeapon's CPA calcs to offset targetPos by -targetVel * (submunitionVelocity / proximityDetonationdist)
```

Comment:
So remove the FramevelocityV3f from pBullet.currentvelocity and just leave it firedVelocity?
This random dispersion velocity would be on top of the potential dispersion velocity addition from the submunition.bulletvelocity during the Gaussian?

Reply:
Yes, but the frame of Krakensbane adjustments I was referring to is adjustments in the position, not in the velocity, but it depends on what the actual source of that offset is.

`BeehiveDetonation` is being called after the parent bullet has been moved, so I think `currPosition` should actually be the correct position for the subprojectiles and the velocities should be the parent bullet's drag-adjusted velocity plus some dispersion.
The subprojectiles are being enabled this frame, so they should get moved on the next frame, which would mean that they shouldn't have missed a frame of Krakensbane adjustments.

What I suspect is happening is that the asymmetry is being caused by the proxy detonation calculation being out of sync with the Flight Integrator which runs in the `TimingManager.TimingStage.FlightIntegrator` stage of `FixedUpdate` (two stages after `TimingManager.TimingStage.Normal`), so the proxy detonation in each case says "I'm close enough, detonate!" during the Normal stage, but then the vessels get moved during the `FlightIntegrator` stage with the main portion of that movement in the direction of the frame velocity so that when it's displayed on the next rendered frame it appears to be off in the same direction by that amount.
If I'm right, then performing the proxy detonation check against each vessel separately with the `OverlapSphere` position shifted back by 1 frame's worth of the vessel's velocity similar to the Vessel-Relative Bullet Checks ought to sync them up again.

Reply:
An alternative that doesn't require multiple `OverlapSphere` calls would be to perform the proxy detonation check before moving the bullet (i.e., on the next frame) and then spawn in the subprojectiles already moved by 1 frame's worth. The detonation visuals would be 1 rendered frame out of sync though.

However, both of these approaches based on `OverlapSphere` suffer another issue, which is that for fast moving vessels (particularly those heading towards the bullet), the proximity check happens in discrete steps and could potentially jump right past a vessel.

Perhaps a better alternative would be to use `AIUtils.TimeToCPA` against all the vessels with `maxTime = TimeWarp.fixedDeltaTime` to detect if the closest approach is within the vessel's radius + proxy detonation distance and then detonate at approximately the time where it crosses that distance (linear interpolation should be fine for such a short time) moving the subprojectiles by the remaining time in the frame (using `CheckBulletCollisions` and `MoveBullet`).

Reply:
An even more accurate way to do this would be to use spherecasts relative to each vessels' velocity frame, but that would be fairly expensive computationally. It could be combined with the `TimeToCPA` method to only check relevant vessels though to reduce the computation. It'd look something like this each frame prior to moving the bullet:
```csharp
period = Time.fixedDeltaTime;
r = det radius;
Check OverlapSphere(r) and detonate if something is already within range;
R = GetOverlapSphereRadius(period);
nearbyVessels = vessels within OverlapSphere(R);
foreach(var vessel in nearbyVessels) {
  if (CPA > r + vessel.GetRadius()) continue;
  maxDist = period * bullet velocity in vessel-relative velocity frame
  check SphereCastAll(r, maxDist) in vessel-relative velocity frame for hits against target vessel;
  if (hit) {
    detonate at detPoint = (hit.point - r * hit.normal);
    dt = period * hit.distance / maxDist;
    spawn subprojectiles at detPoint and move them by period - dt, checking for hits;
  }
}
```

Reply:
Ah, teach me to refresh the page more often, I missed this latest comment and did a pure T2CPA approach, which seems to be working, though I am running into an issue where shots fired from more than ~540m result in it detonating at 2x the detonationRange; you have more experience with the CPA stuff than I, so it's probably some obvious oversight on my part.
Would be interested to see what the performance differences between a bunch of CPA math vs a spherecast per frame per bullet is as well.